pkg/downstream/client.go

```diff
@@ -0,0 +1,145 @@
+package downstream
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type Client struct {
```

Comment:
Added a client to allow us to do client allocation as a one off, rather than in the hotpath.